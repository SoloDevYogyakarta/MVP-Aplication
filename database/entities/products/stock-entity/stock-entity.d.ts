import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class ProductStockEntity extends BaseEntity {
    use_stock?: boolean;
    value?: number;
    stock_wording?: number;
    product_id?: string;
}
export type ProductStockCreationAttribute = Optional<ProductStockEntity, 'id'>;
export interface ProductStockInstance extends Model<ProductStockCreationAttribute, ProductStockEntity>, ProductStockEntity {
}
export declare const productStockEntity: import("sequelize").ModelCtor<ProductStockInstance>;
