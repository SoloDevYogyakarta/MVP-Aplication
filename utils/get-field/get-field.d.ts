import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
