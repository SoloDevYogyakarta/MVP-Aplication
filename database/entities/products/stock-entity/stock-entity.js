"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.productStockEntity = exports.ProductStockEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class ProductStockEntity extends base_entity_1.BaseEntity {
}
exports.ProductStockEntity = ProductStockEntity;
exports.productStockEntity = entity_1.sequelize.define('STOCK', {
    public_id: sequelize_1.DataTypes.STRING,
    use_stock: sequelize_1.DataTypes.BOOLEAN,
    value: sequelize_1.DataTypes.INTEGER,
    stock_wording: sequelize_1.DataTypes.INTEGER,
    product_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'STOCK',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=stock-entity.js.map