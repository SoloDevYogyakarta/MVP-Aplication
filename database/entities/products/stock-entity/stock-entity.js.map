{"version":3,"file":"stock-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/products/stock-entity/stock-entity.ts"],"names":[],"mappings":";;;AAAA,mCAAgC;AAChC,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,kBAAmB,SAAQ,wBAAU;CAKjD;AALD,gDAKC;AAMY,QAAA,kBAAkB,GAAG,kBAAS,CAAC,MAAM,CAChD,OAAO,EACP;IACE,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,SAAS,EAAE,qBAAS,CAAC,OAAO;IAC5B,KAAK,EAAE,qBAAS,CAAC,OAAO;IACxB,aAAa,EAAE,qBAAS,CAAC,OAAO;IAChC,UAAU,EAAE,qBAAS,CAAC,MAAM;CACY,EAC1C;IACE,SAAS,EAAE,OAAO;IAClB,MAAM,EAAE,UAAU;IAClB,UAAU,EAAE,KAAK;IACjB,KAAK,EAAE;QACL,YAAY,CAAC,UAAU,EAAE,OAAO;YAC9B,MAAM,QAAQ,GAAG,UAAU,CAAC;YAC5B,QAAQ,CAAC,SAAS,GAAG,IAAA,eAAM,GAAE,CAAC;QAChC,CAAC;KACF;CACF,CACF,CAAC"}