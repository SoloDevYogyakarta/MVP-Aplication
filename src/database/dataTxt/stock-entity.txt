{"id":169,"use_stock":true,"value":38,"stock_wording":7,"product_id":"nusX-9TrZDOiVWTGhF2Cd","public_id":"xyUXDD7ajAAkD28w8fBul"}