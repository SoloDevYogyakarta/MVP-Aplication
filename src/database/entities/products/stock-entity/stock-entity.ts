import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ProductStockEntity extends BaseEntity {
  use_stock?: boolean;
  value?: number;
  stock_wording?: number;
  product_id?: string;
}

export type ProductStockCreationAttribute = Optional<ProductStockEntity, 'id'>;
export interface ProductStockInstance
  extends Model<ProductStockCreationAttribute, ProductStockEntity>,
    ProductStockEntity {}
export const productStockEntity = sequelize.define<ProductStockInstance>(
  'STOCK',
  {
    public_id: DataTypes.STRING,
    use_stock: DataTypes.BOOLEAN,
    value: DataTypes.INTEGER,
    stock_wording: DataTypes.INTEGER,
    product_id: DataTypes.STRING,
  } as ModelAttributes<ProductStockInstance>,
  {
    tableName: 'STOCK',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
