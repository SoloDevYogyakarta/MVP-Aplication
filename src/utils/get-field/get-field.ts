import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
