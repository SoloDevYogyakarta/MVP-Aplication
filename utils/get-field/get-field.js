"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map