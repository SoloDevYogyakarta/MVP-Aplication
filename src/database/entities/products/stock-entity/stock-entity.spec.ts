import { faker } from '@faker-js/faker';
import { ModelCtor } from 'sequelize';
import { getField } from '../../../../utils/get-field/get-field';
import { productStockEntity, ProductStockInstance } from './stock-entity';

describe('ProductStockEntity', () => {
  let entity: ModelCtor<ProductStockInstance>;

  beforeEach(() => {
    entity = productStockEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('update', async () => {
      const { public_id } = getField('stock-entity');
      const findOne = await entity.findOne({ where: { public_id } });
      const value = faker.number.int({ min: 10, max: 20 });
      findOne.value = value;
      findOne.save();
      expect(findOne.value).toEqual(value);
    });
    it('findOne', async () => {
      const { public_id } = getField('stock-entity');
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  } catch (er) {
    // empty
  }
});
