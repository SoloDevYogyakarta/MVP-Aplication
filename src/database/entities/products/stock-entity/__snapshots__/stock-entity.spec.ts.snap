// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductStockEntity render correctly 1`] = `[Function]`;
